raiden/network/rpc/client.py

```diff
@@ -439,11 +439,17 @@ def _query_filter(self, function):
                 decode_topic(topic)
                 for topic in log_event['topics']
             ]
+            block_number = log_event.get('blockNumber')
```

Comment:
This is what I meant you should do. The filer should be queried from [this RPC call](https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getfilterchanges) and should almost always have a block. IF there is no `blockNumber` attribute, as you can see from the documentation that means it's the pending block though, not `0`.